lld/ELF/Driver.cpp

```diff
@@ -1750,7 +1797,14 @@ void LinkerDriver::createFiles(opt::InputArgList &args) {
       break;
     }
     case OPT_script:
-      if (std::optional<std::string> path = searchScript(arg->getValue())) {
+      // EraVM local begin
+      if (config->useIOMemoryBuffers) {
+        if (std::optional<MemoryBufferRef> mb = readFile(arg->getValue()))
```

Comment:
You are right, we do not need `searchScript` when LLD In/Out is performed via memory buffers. Instead, we call   `readFile()` that contains a bit hackery code that instead of reading a file on OS just returns the reference to the passed memory buffer:
  ```
  ...
MemoryBufferRef mbref;
  if (config->useIOMemoryBuffers) {
    unsigned idx = 0;
    if (path.getAsInteger(10, idx)) {
      error(path + ": path should be an index");
      return std::nullopt;
    }
    if (idx >= config->inBuffers.size()) {
      error("memory buffer index is out of range");
      return std::nullopt;
    }
    mbref = config->inBuffers[idx];
    ...
```
where `arg->getValue()` is the memory buffer index, "0", or "1", so on.
Please, check the changes in `readFile()` in lld/ELF/InputFiles.cpp, line 234.